test/test_label_drop.py

```diff
@@ -13,36 +13,73 @@ class ConstraintAndIndex(StructuredNode):
 
 
 def test_drop_labels():
-    constraints_before, meta = db.cypher_query("CALL db.constraints()")
-    indexes_before, meta = db.cypher_query("CALL db.indexes()")
+    # From Neo4j 4.3 on, the syntax to list constraints is 'SHOW CONSTRAINTS'
+    # For older versions, this returns a CypherSyntaxError, and the correct syntax is 'CALL db.constraints()'
+    # Same applies for indexes
+    try:
+        constraints_before, meta_constraints_before = db.cypher_query("SHOW CONSTRAINTS")
+        indexes_before, meta_indexes_before = db.cypher_query("SHOW INDEXES")
+    except CypherSyntaxError:
+        constraints_before, meta_constraints_before = db.cypher_query("CALL db.constraints()")
+        indexes_before, meta_indexes_before = db.cypher_query("CALL db.indexes()")
 
     assert len(constraints_before) > 0
     assert len(indexes_before) > 0
 
     remove_all_labels()
 
-    constraints, meta = db.cypher_query("CALL db.constraints()")
-    indexes, meta = db.cypher_query("CALL db.indexes()")
+    try:
+        constraints, meta_constraints = db.cypher_query("SHOW CONSTRAINTS")
+        indexes, meta_indexes = db.cypher_query("SHOW INDEXES")
+    except CypherSyntaxError:
+        constraints, meta_constraints = db.cypher_query("CALL db.constraints()")
+        indexes, meta_indexes = db.cypher_query("CALL db.indexes()")
 
     assert len(constraints) == 0
+    indexes_as_dict = [dict(zip(meta_indexes, row)) for row in indexes]
+    constraints_before_as_dict = [dict(zip(meta_constraints_before, row)) for row in constraints_before]
+    indexes_before_as_dict = [dict(zip(meta_indexes_before, row)) for row in indexes_before]
     # Ignore the automatically created LOOKUP indexes
-    assert len([index for index in indexes if index[7] != []]) == 0
+    assert len([index for index in indexes_as_dict if index["labelsOrTypes"]]) == 0
 
-    # Returning all old constraints and indexes
-    # Versions prior to 4.0 have a very different return format
-    for constraint in constraints_before:
-        if constraint[0].startswith('CONSTRAINT '):
-            db.cypher_query('CREATE ' + constraint[0])
+    # Recreating all old constraints and indexes
+    # Versions before 4.3 had a 'description' column which can be used to recreate the constraint
+    for constraint in constraints_before_as_dict:
+        if "description" in constraint:
+            db.cypher_query('CREATE ' + constraint["description"])
         else:
-            db.cypher_query('CREATE ' + constraint[1])
-    for index in indexes_before:
+            constraint_type_clause = "UNIQUE"
+            if constraint["type"] == "NODE_PROPERTY_EXISTENCE":
+                constraint_type_clause = "NOT NULL"
+            elif constraint["type"] == "NODE_KEY":
+                constraint_type_clause = "NODE KEY"
+            try:
+                db.cypher_query('CREATE CONSTRAINT {0} FOR (n:{1}) REQUIRE n.{2} IS {3}'
```

Comment:
This would definitely read better as an f-string (Nothing to be done now, just a note for future reference)